### ISSUE TITLE
A needless route segment added elsewhere when inserting an intermediate destination

### ISSUE THREAD

**Issue by Kjarrval** [osmandapp/Osmand#9516]
The best way to get help about an OsmAnd issue is to create a valid and detailed issue content.
Please give us the following information so that we can try to **reproduce** your issue:
**1. Add a destination (by car) from Kirkjubæjarklaustur (63.79523° N, 18.03874° W) to Djúpivogur (64.65883° N, 14.28404° W)
2. Add an intermediate destination to Hoffellsjökull (64.43272° N, 15.44238° W)**

What version of OsmAnd are you using, on what device and Android/iOS version?
**OsmAnd+ 3.7.4, released 2020-06-09
Galaxy S8, Android version 9**

Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs, and its edition date.
**Offline map of Iceland downloaded within OsmAnd, with live update timed 25 July 2020 17:28**

__*In case of wrong routing:*__

Tell us whether you have used OsmAnd's in-app offline routing, or any online routing provider like YOURS, OpenRouteService or OSRM.
**The in-app offline routing.**

What routing profile is chosen in OsmAnd app? (car, bike, pedestrian, fastest or shortest)
**Car, both in fastest and shortest.**

Please specify as exactly as possible the start and end point of your route by telling us city name and street name so that we can find it via in-app address search easily. Also, a permalink from openstreetmap.org can be helpful.
**See coordinates in a previous answer.**

Tell us your expected routing, and how OsmAnd routes, or add screenshots here.
**Expected routing is without the additional road segment.**

**The problem:
An additional and pointless back and forth route is generated from the crossing of Austurlandsvegur and Kollumúlavegur (roughly 64.40787° N, 14.90761° W) along Kollumúlavegur until around 64.47592° N, 15.09378° W
Removal of the intermediate destination will remove the additional route segment**

**Note 1: That this will not happen if the intermediate destination is a little bit more south on Hoffellsjökull
Note 2: If the coordinates on the intermediate destination are the starting position, the route will begin by "jumping" to Kollumúlavegur**

**Comment by scaidermern** [comment]
Can you add a screenshot? This sounds like #3582.

**Comment by Kjarrval** [comment]
> Can you add a screenshot? This sounds like #3582.

![Screenshot_20200727-080631_OsmAnd+](https://user-images.githubusercontent.com/2019152/88518865-b5ec2f00-cfe0-11ea-900f-3495e19ac6de.jpg)

Note that there are much closer roads (not displayed in this zoom) directly south of the intermediate destination which would make more sense to suggest than going over a mountain.

**Comment by vshcherb** [comment]
Not sure probably it is fixed already with 3.8 nightly builds and screenshot doesn't show any wrong roads.

**Comment by Kjarrval** [comment]
> Not sure probably it is fixed already with 3.8 nightly builds and screenshot doesn't show any wrong roads.

![88518865-b5ec2f00-cfe0-11ea-900f-3495e19ac6de](https://user-images.githubusercontent.com/2019152/88543372-5ead8500-d007-11ea-9117-e39f6c8426b7.jpg)

The red circled area is the wrong road. A much more sensible suggestion would've been in the blue-coloured area.

**Comment by vshcherb** [comment]
Yes it is a tricky situation. I agree with the picture above it would be more logical.
P.S. The distance from point to the nearest road seems to be more than 3-5 km (https://www.openstreetmap.org/node/2419266854#map=11/64.4467/-15.3088)
